Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -569,6 +569,220 @@ class UpstreamIntegrationTests: TestBase {
         assertExpectedEvents(ignoreUnexpectedEvents: true)
     }
 
+    // Tests datastream config overrides
+    func testSendEvent_withValidConfigOverrides_receivesExpectedNetworkResponse() {
+        // Setup
+        // Note: test constructs should always be valid
+        let interactNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint), httpMethod: .post)!
+        // Setting expectation allows for both:
+        // 1. Validation that the network request was sent out
+        // 2. Waiting on a response for the specific network request (with timeout)
```

Comment:
nit: let's remove these notes from tests, the intent is covered by using the corresponding framework APIs